libnmstate/nm/connection.py

```diff
@@ -217,10 +234,22 @@ def _active_connection_callback(self, src_object, result, user_data):
             )
 
             ac = ActiveConnection(nm_act_con)
+            nm_acsreason = nmclient.NM.ActiveConnectionStateReason
             if ac.is_active:
+                self._activate_retry_count = 0
                 self._mainloop.execute_next_action()
             elif ac.is_activating:
                 self.waitfor_active_connection_async(ac)
+            elif ac.reason == nm_acsreason.DEVICE_DISCONNECTED:
```

Comment:
After commenting in the `waitfor` method, I am just wandering if we just need to call waitfor and not retry activation.
So we could add to the previous condition something like: `elif ac.is_activating or ac.reason == nm_acsreason.DEVICE_DISCONNECTED:`

Reply:
When device deactivated, its activate connection will be None. There will be no `state-changed` signals.